### ISSUE TITLE
[indent-headings] export to pdf issue

### ISSUE THREAD

**Issue by sdaognklwb** [zamsyt/obsidian-snippets#16]
Hello @zamsyt 

A really big thank you for making this snippets. It has made my day. 

The plugin works exceedingly well so far. However, I am just wondering if it is intended for the indents not to be carried over to the pdf export? AKA I can see the indents in preview mode fine, but when I export them to a pdf file, they fall back to how they are originally placed, rather than indented. 

Thanks and looking forward to hearing from you.

**Comment by zamsyt** [comment]
Thanks for the kind words! @sdaognklwb

The snippet needs separate rules for the different modes due to them being different, and I just haven't added that for the PDF exports before. I've now added that support in the dev branch: [dev/Indent headings.css](https://github.com/zamsyt/obsidian-snippets/blob/dev/Indent%20headings.css). Let me know if it's working as expected.

I'll wait for a while before pushing this change in the main version (might do a performance comparison myself). This snippet requires a large set of complicated rules to work---once over for each of the now 3 rule sets! I've had to fix a performance issue before, and if this change ends up having a noticeable impact, I might consider providing the version with PDF support as a separate version.

**Comment by sdaognklwb** [comment]
Hi @zamsyt 

Thanks for the update. It works wonder so far with the pdf export. Thank you so much. 

I have another question regarding how the headings are structured. 

So from the below screenshot, you can see the first sentence and first-level heading are aligned. However, after I used a second-level header, all the content that I subsequently typed will be under this second-level heading. I will not be able to start a new sentence that does not belong to any heading afterwards. Is this an intended effect?

In simpler terms, what I want to do is if i want to type a new sentence that aligns with "This is an apple" after "This is a carrot", is it possible? Currently, it is impossible and everything i type after "this is a carrot" will be under "this is a carrot", and I cannot open a new line that does not belong to that. 

 I know it sounds confusing, but I hope you understand. 

THANK YOUUU UwU

<img width="322" alt="Screenshot 2024-05-02 at 10 41 34" src="https://github.com/zamsyt/obsidian-snippets/assets/100029725/2d5a2c29-10e7-4ebe-a1ff-b31bd1ab316f">

**Comment by zamsyt** [comment]
@sdaognklwb With the snippet, I try to make the indentation match the way Obsidian itself sees the heading hierarchy, e.g. when folding headings:

https://github.com/zamsyt/obsidian-snippets/assets/102934832/3e0a26f5-8b28-439f-98f8-f3afa1f8c038

As far as Obsidian itself is concerned (when it comes to folding, or heading embeds), there's no way to "escape" a heading without adding a new heading after it. One convention used by some people is to use empty headings for this purpose (just some number of `#` symbols without text after them), and that will work with my snippet as well. It should be technically possible for me to add a separate way to do this escaping in my snippet, but that wouldn't be recognized by any other features of Obsidian, and I don't see a solution that would be an improvement on just using empty headings.


---

Some side notes:

My snippet sees top level text, and text under h1's as different categories, but I added an offset by default that places h1 text aligned with the top level text (instead of being indented) since h1's are commonly used as a title for the whole document, and I didn't want to indent the whole thing because of it. You can play around with these offsets using the variables on [lines 5 and 6](https://github.com/zamsyt/obsidian-snippets/blob/3b8fa6f6f9c9bca778161201c67c8ad63246b3b8/Indent%20headings.css#L5).

In your screenshot, you only had one paragraph of text after every heading. I'm guessing that's not true in your real use, but if in fact you only ever wanted to indent the first paragraph below every heading, a much simpler snippet could be created to do exactly that, without the limitations of this snippet.

**Comment by sdaognklwb** [comment]
Hello @zamsyt 

Thank you so much for the detailed response. It certainly mirrors my experience of using obsidian. I am just wondering if the following is possible however. 

# This is a banana (header 1) 
This is a banana lalala (para 1) 

  ## This is a carrot (header 2)
        This is a carrot (para 1)

This is a banana lala (para 2 of header 1)

in simpler terms, can I continue writing a paragraph for, and under header 1, after the creation of header 2?

My understanding is that it cannot in original obsidian. But do you think think there is a way around this? I know it is perhaps too much to ask and i really appreciate your patience. 

Thank youu <3333

**Comment by zamsyt** [comment]
Oh, I think I understand now. If I understand correctly, this describes what you'd want, but you'd want the last paragraph shifted to the left:
![image](https://github.com/zamsyt/obsidian-snippets/assets/102934832/04d58dc2-9627-4315-9a37-c4f82942a5af)
Doing it like above would be the conventional way to do it without this snippet, but I haven't realized before that with this snippet, it won't look like what you'd want.

**Comment by sdaognklwb** [comment]
Hi @zamsyt 

Thanks for the quick response. I think you have described accurately on what I wanted, and just to confirm - you are saying it will not be possible with the snippet?

<img width="285" alt="Screenshot 2024-05-04 at 14 27 23" src="https://github.com/zamsyt/obsidian-snippets/assets/100029725/6f1bab3c-a375-4e73-90a2-c01a7431e088">

Thanks a lot!

**Comment by zamsyt** [comment]
@sdaognklwb I'm thinking about it. I think this functionality would indeed be a nice addition to the snippet. Here's already a quick test version that should make the empty heading syntax work in reading view: [Indent headings_esc-test.css](https://gist.github.com/zamsyt/369973d7b4eaf0c95bc8912d90c0c06f). However, in edit mode, I don't think it's possible the same way with CSS alone. (Could also be possible by manually adding special marker elements and targeting those in the snippet, but that's not ideal). If there was a plugin similar to Contextual Typography that added the selector I need in edit mode, I might e.g. consider adding the edit mode support targeting such a plugin. Hmm. Is just the reading view feature of use to you? And at least if it were possible to make work the same way in editing mode, would you find that a good solution?

**Comment by sdaognklwb** [comment]
Hi @zamsyt. Thanks for getting back so quickly.

If it is possible to mirror the performance in edit mode and reading mode in the same way - that would be marvellous. Please kindly take your time!

I will give the new css a try and let you know how it goes. 

Thank you :)

**Comment by zamsyt** [comment]
Here's also a second test version that shows how the special marker element could work: [Indent headings_esc-test2.css ](https://gist.github.com/zamsyt/50fcfeea932325a42e82ee3809112b1d). This works in both edit and reading modes, but usage is using inline html like this:
```md
# h1
text

## h2
text

## <span class="heading-escape"></span>
text
```
Not as elegant. I probably don't want to integrate a solution like this into the main snippet. But if someone really wants, I could perhaps consider providing a solution like this as a separate snippet.

**Comment by sdaognklwb** [comment]
Hello @zamsyt 

Thanks for the reply. I have tried the above .css and as you said, they worked but may not be the most elegant solution, but I am very grateful. 

A separate snippet will be lovely. But please kindly take your time and at your own pace. I am sure such snippets are highly desirable among Obsidian members, especially those who are not used to the classic .md framework (myself included).  It doesn't necessarily need to be incorporated into the main snippet, but a separate snippet that will be compatible with the main will be amazing.

I am very grateful towards your assistance so far. Thank you so very much :)
